cluster.tf

```diff
@@ -3,22 +3,33 @@ resource "google_container_cluster" "primary" {
   name     = "${var.project_id}-cluster${var.cluster_name_suffix}"
   location = local.location
   provider = google-beta
-  # So we create the smallest possible default node pool and immediately delete it. Using node pools
+
+  # We can't create a cluster with no node pool defined, but we want to only use separately managed node pools.
+  # So we create the smallest possible default node pool and immediately delete it.
   remove_default_node_pool = true
   initial_node_count       = 1
-  min_master_version       = var.kubernetes_version
-  node_version             = var.kubernetes_version
+
+  min_master_version = local.master_version
+
   # Point to stackdriver api
   monitoring_service = "monitoring.googleapis.com/kubernetes"
   logging_service    = "logging.googleapis.com/kubernetes"
   network            = google_compute_network.network.self_link
   subnetwork         = google_compute_subnetwork.subnetwork.self_link
 
+  release_channel {
+    channel = var.release_channel
+  }
+
   private_cluster_config {
-    enable_private_nodes   = true
-    master_ipv4_cidr_block = var.master_ipv4_cidr_block
+    enable_private_endpoint = false
+    enable_private_nodes    = true
+    master_ipv4_cidr_block  = var.master_ipv4_cidr_block
   }
 
+  # TODO: Shouldn't we set this property?
+  # https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/container_cluster#networking_mode
+  networking_mode = "VPC_NATIVE"
```

Comment:
I've read that if you define the `ip_allocation_policy` as we do in this file this is taken as default `VPC_NATIVE`. Nevertheless, I think it won't hurt if we keep it explicitly like this.

Reply:
Agree with Santi, we can keep it for the sake of clarity.
We just need to remove the TODO comment